### DISCUSSION TITLE
winhello and keys on iPad/iPhone

### DISCUSSION THREAD

**Discussion by gkodinov** [Yubico/libfido2#803] [Q&A]
Hello,

I'm writing a program that registers a passkey in WinHello and then uses it to authenticate.

After enabling winhello in the build settings I've been able to successfully register a passkey and authenticate against it when I am storing the passkey into my windows store.

But if I try to pick the ipad/iphone option from the WinHello dialog and then try to scan the QR code with my iphone I get HERROR hr=0x800704c7 (ERROR_CANCELED) from the WebAuthNAuthenticatorMakeCredential().

I then tried the libfido2's native command line tools (fido2-cred in specific) but that doesn't even get me the WinHello dialog.

I'd appreciate any insights in how you register a passkey on an iPhone from a windows device via libfido2 APIs.

**Comment by gkodinov** [comment]
![Untitled](https://github.com/Yubico/libfido2/assets/4089734/0c8203a0-9da3-4048-aaa7-bf99200c6dba)

**Comment by LDVG** [comment]
Hi,

Support for the hybrid transport via the libfido2 `windows://hello` pseudo device has not yet been tested or verified by us. We'll let you know once we know more.

**Reply by gkodinov** [reply]
Thanks! Is there any way to instruct libfido to not show the unsupported options in the WinHello dialogs then?

**Reply by LDVG** [reply]
I'm unfortunately not aware of any such API (see webauthn.h). As an aside, I would have assumed the Windows Hello interface to abstract mostly everything about the hybrid transport, and it seems strange to me that it's failing for you.
